R/model_diagnostics.R

```diff
@@ -18,24 +21,46 @@
 #' @param rhat_tolerance float indicating the tolerance for the rhat for
 #' individual parameters, default is `1.05`
 #' @param max_tree_depth_tol float indicating the tolerance for the proportion
-#' of iterations that exceed the maximum tree depth, default is `0.01`
+#' of iterations that exceed the maximum tree depth, default is `0.01`,
+#' @param ... additional arguments
 #'
+#' @family diagnostics
 #' @return flag_df: dataframe containing columns for each of the flags,
 #' if any flags are TRUE that indicates some model issue
 #' @export
+get_model_diagnostic_flags <- function(x, ...) {
+  UseMethod("get_model_diagnostic_flags")
+}
+
+#' S3 method for getting a table of diagnostic flags fpr a wwinference_fit
+#' object
+#'
+#' This method overloads the generic get_model_diagnostic_flags function
+#' specifically for objects of type 'wwinference_fit'.
 #'
-get_model_diagnostic_flags <- function(stan_fit_object,
-                                       ebmfi_tolerance = 0.2,
-                                       divergences_tolerance = 0.01,
-                                       frac_high_rhat_tolerance = 0.05,
-                                       rhat_tolerance = 1.05,
-                                       max_tree_depth_tol = 0.01) {
-  n_chains <- stan_fit_object$num_chains()
-  iter_sampling <- stan_fit_object$metadata()$iter_sampling
+#' @rdname get_model_diagnostic_flags
+#' @export
+get_model_diagnostic_flags.wwinference_fit <- function(x, ...) {
+  get_model_diagnostic_flags.default(
+    x = x$fit$result
+  )
+}
+
+#' @rdname get_model_diagnostic_flags
+#' @export
+get_model_diagnostic_flags.default <- function(x,
```

Comment:
is this really a default method or is it a cmdstan method?

Reply:
This is a custom function that we set up when we made the pipeline. It uses diagnostic thresholds that we set as defaults (user can alter them), and returns TRUE if the threshold is exceeded.

Reply:
no sorry this is the same question as for the other s3 method i.e should this only dispatch if x is a CMDSTANMODEl (lol I don't think that is actually the class name).